core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java

```diff
@@ -0,0 +1,197 @@
+   * rewritten. This is considered in conjunction with min-num-input-files, both
+   * conditions must pass to consider a group of files to be rewritten.
+   */
+  public static final String MIN_OUTPUT_FILES = "min-output-files";
+  public static final int MIN_OUTPUT_FILES_DEFAULT = 1;
+
+  /**
+   * Adjusts files which will be considered for rewriting. Files smaller than
+   * min-file-size-bytes will be considered for rewriting. This functions independently
+   * of max-file-size-bytes.
+   * <p>
+   * Defaults to 75% of the target file size
+   */
+  public static final String MIN_FILE_SIZE_BYTES = "min-file-size-bytes";
+  public static final double MIN_FILE_SIZE_DEFAULT_RATIO = 0.75d;
+
+  /**
+   * Adjusts files which will be considered for rewriting. Files larger than
+   * max-file-size-bytes will be considered for rewriting. This functions independently
+   * of min-file-size-bytes.
+   * <p>
+   * Defaults to 180% of the target file size
+   */
+  public static final String MAX_FILE_SIZE_BYTES = "max-file-size-bytes";
+  public static final double MAX_FILE_SIZE_DEFAULT_RATIO = 1.80d;
+
+  private int minInputFiles;
+  private int minOutputFiles;
+  private long minFileSize;
+  private long maxFileSize;
+  private long targetFileSize;
+  private long maxGroupSize;
+
+  @Override
+  public String name() {
+    return Strategy.BINPACK.name();
+  }
+
+  @Override
+  public Set<String> validOptions() {
+    return ImmutableSet.of(
+        MIN_INPUT_FILES,
+        MIN_OUTPUT_FILES,
+        MIN_FILE_SIZE_BYTES,
+        MAX_FILE_SIZE_BYTES
+    );
+  }
+
+  @Override
+  public RewriteStrategy options(Map<String, String> options) {
+    targetFileSize = PropertyUtil.propertyAsLong(options,
+        RewriteDataFiles.TARGET_FILE_SIZE_BYTES,
+        PropertyUtil.propertyAsLong(
+            table().properties(),
+            TableProperties.WRITE_TARGET_FILE_SIZE_BYTES,
+            TableProperties.WRITE_TARGET_FILE_SIZE_BYTES_DEFAULT));
+
+    minFileSize = PropertyUtil.propertyAsLong(options,
+        MIN_FILE_SIZE_BYTES,
+        (long) (targetFileSize * MIN_FILE_SIZE_DEFAULT_RATIO));
+
+    maxFileSize = PropertyUtil.propertyAsLong(options,
+        MAX_FILE_SIZE_BYTES,
+        (long) (targetFileSize * MAX_FILE_SIZE_DEFAULT_RATIO));
+
+    maxGroupSize = PropertyUtil.propertyAsLong(options,
+        RewriteDataFiles.MAX_FILE_GROUP_SIZE_BYTES,
+        RewriteDataFiles.MAX_FILE_GROUP_SIZE_BYTES_DEFAULT);
+
+    minOutputFiles = PropertyUtil.propertyAsInt(options,
+        MIN_OUTPUT_FILES,
+        MIN_OUTPUT_FILES_DEFAULT);
+
+    minInputFiles = PropertyUtil.propertyAsInt(options,
+        MIN_INPUT_FILES,
+        MIN_INPUT_FILES_DEFAULT);
+
+    validateOptions();
+    return this;
+  }
+
+  @Override
+  public Iterable<FileScanTask> selectFilesToRewrite(Iterable<FileScanTask> dataFiles) {
+    return FluentIterable.from(dataFiles)
+        .filter(scanTask -> scanTask.length() < minFileSize || scanTask.length() > maxFileSize);
```

Comment:
Currently they are synonymous, BaseFileScanTask just calls file.length. I'm not super worried about this just yet. I think this won't be an issue in the future issue since you can't determine how many records are deleted and the new scan task size without actually reading the file so I don't think that info will be there?

Reply:
I was worried we could add the size of deletes files. I checked the doc for this method and I don't think that's going to happen. I think we are good here.